core/communication_manager.py

```diff
@@ -0,0 +1,155 @@
+        }
+        logging.info("Communication Manager is set up!")
+
+    # Setup Methods
+
+    def configure(self, scheduler):
+        """
+        Configures the Communication Manager so that it can submit jobs to the
+        Scheduler.
+        """
+        logging.info("Configuring the Communication Manager...")
+        self.scheduler = scheduler
+        logging.info("Communication Manager is configured!")
+
+    # Public methods
+
+    def inform(self, event_type, payload):
+        """
+        Method called by other modules to inform the Communication Manager about
+        events that are going on in the service.
+
+        If there isn't an active optimizer, the commmgr will handle it
+
+        Else, payloads are relayed to the active optimizer which decides how
+        and to whom the Communication Manager should communicate the event/message.
+
+        For example: A runner could inform the Communication Manager that the
+        node is done training a particular model, to which the Optimizer could
+        decide it's time to communicate the new weights to the network.
+        """
+        logging.info("Information has been received: type({})".format(event_type))
+        if self.optimizer:
+            # We have an active session.
+            event_type, payload = self.optimizer.ask(event_type, payload)
+            callback = callback_handler(
+                event_type,
+                self.ACTIONABLE_EVENT_TYPE_2_CALLBACK,
+                ActionableEventTypes.NOTHING.value
+            )
+        else:
+            # We need to start a new session.
+            callback = callback_handler(
+                event_type,
+                self.RAW_EVENT_TYPE_2_CALLBACK,
+                RawEventTypes.NOTHING.value
+            )
+        callback(payload)
+
+    # Callbacks
+
+    def _create_session(self, payload):
+        """
+        Creates a new session and optimizer based on the parameters sent by a
+        model developer. It then asks the optimizer what's the first thing to do
+        and the service starts working on that session from that point on.
+        """
+        # NOTE: We removed the 'optimizer_type' argument since for the MVP we're
+        # only considering the 'FederatedAveragingOptimizer' for now.
+        # TODO: We need to incorporate session id's when we're ready.
+        logging.info("New optimizer session is being set up...")
+        content = payload.get('content')
+        self.optimizer = FederatedAveragingOptimizer(content)
+        logging.info("Optimizer session is set! Now doing kickoff...")
+        event_type, payload = self.optimizer.kickoff()
+        callback = callback_handler(
+            event_type,
+            self.ACTIONABLE_EVENT_TYPE_2_CALLBACK,
+            ActionableEventTypes.NOTHING.value
+        )
+        callback(payload)
+        logging.info("Kickoff complete.")
+
+    def _transform_and_split(self, payload):
```

Comment:
If this hasn't been implemented yet, can you remove this method or comment it out?

Reply:
I think some things would fail if we remove, because the Listener assumes this type of job exists. It's not a biggie for now since you'll be done implementing the functionality to make this work pretty soon.